src/main/scala/fm/http/server/HttpServerApp.scala

```diff
@@ -325,13 +325,14 @@ abstract class HttpServerApp extends Logging {
    * Note: USR1 is already being used by the JVM for something:
    *  "java.lang.IllegalArgumentException: Signal already used by VM or OS: SIGUSR1"
    */
-  private def registerPingSignalHandlers(server: HttpServer): Unit = {
+  private def registerPingSignalHandlers(server: HttpServer): Unit = try {
     Signal.handle(new Signal("USR2"), new SignalHandler{
       def handle(signal: Signal): Unit = {
         logger.info("Caught USR2 Signal - Disabling /_ping")
         server.disablePing()
       }
     })
+  } catch {
+    case ex: Throwable => // Cannot register signal handlers (on Windows, or sun.misc.Signal unavailable)
```

Comment:
If the signal cannot be registered then you can also hit this exception handler in which case it's silently ignored.  That doesn't seem ideal...

Reply:
For the catch, could put something like ```
val isWin = System.getProperty("os.name", "") startsWith "Windows"
if (!isWin) throw ex
```

I haven't looked into the whole reason you're using the signals, but was there a reason you're not just hooking into `Runtime.getRuntime().addShutdownHook(...`

Reply:
Shutdown Hooks get run when everything is shutting down.  By then it's too late to gracefully disable the ping and wait for the instance to be removed from the load balancer.